### PR TITLE
Data Distributor Plugin

### PR DESCRIPTION
**Pull request by sveseli**
This PR introduces data distributor plugin for the pvDatabase server.  Modifications of the existing code include a single line change to one of the source makefiles, and additional tests in the plugin test code.

### CODE REVIEW

**Review by anjohnson — APPROVED** [review]
Inline comments about the documentation below, I haven't actually tested the code myself or looked too closely at it (your trim functions might use the `find_first_not_of()` and `find_last_not_of()` methods to potentially be slightly more efficient, but there aren't likely to be many spaces here anyway so the difference is probably unmeasurable).

I see that the Codacy checks had some suggestions for performance improvements but I'm not sure whether they would be correct for this code. Please take a look and comment either way, but again probably unmeasurable anyway.

**Comment by sveseli** [comment]
As far as codacy checks, they are all the same: "Inefficient usage of string::find() in condition; string::starts_with() could be faster." As far as I understand, suggested change is C++20, so I would rather leave things as they are. Given that all of those are for parsing the pv request string (done only when establishing client connection), I do not think this would matter at all as far as performance.

File: documentation/dataDistributorPlugin.md [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+The underscore character at the begining of the PV request object
+indicates that the data distributor will be targeting entire PV structure.
+The same PV request object format should work regardless of the language
+in which a particular client application is written.
+
+The plugin parameters are the following:
+
+- group: this parameter indicates a group that client application belongs to
+(default value: "default"); groups of clients are completely independent
+of each other
```
[the remark this answers is not in the document]
Comment (sveseli) [reply documentation/dataDistributorPlugin.md]: Done.

File: documentation/dataDistributorPlugin.md [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+  - one: update goes to one client per set
+  - all: update goes to all clients in a set
+  - default is "one" if client set id is not specified, and "all" if set 
+    id is specified
+
+The plugin obeys the following rules:
+
+- Parameter names are case insensitive, but the string values
+are not. For example, "group=abc" and "group=ABC" would indicate two
+different groups of clients.
```
[the remark this answers is not in the document]
Comment (sveseli) [reply documentation/dataDistributorPlugin.md]: Clarified allowed characters.

**Review by anjohnson — APPROVED** [review]
Looks good, thanks!

**Pull request merged**